File: python_programming/chapter4/word_jumble.py
import random

WORDS = ("python", "jumble", "easy", "difficult", "answer", "xylophone")
HINT = ("snake", "game name", "not difficult", "not easy", "AI", "never")

choice = random.randrange(len(WORDS))
word = WORDS[choice]
hint = HINT[choice]

correct = word

jumble = ""

while word:
	position = random.randrange(len(word))
	jumble += word[position]
	word = word[:position] + word[(position + 1):]

print("\nThe jumble is: ", jumble)
guess = input("\nYour guess: ")
while guess != correct and guess != " ":
	if guess == "hint":
		print(hint)
		guess = input("Your guess:")
	else:
		print("Sorry, that's incorrect.")
		guess = input("Your guess: ")



if guess == correct:
	print("Correct!\n")


input("\nPress enter to exit.")

File: python_programming/chapter4/word_jumble.rb

word = ["python", "jumble", "easy", "difficult", "answer", "xylophone"]
hint = ["snake", "game name", "not difficult", "not easy", "AI", "never"]

choice = rand(word.length())
word_choice = word[choice]
hint_choice = hint[choice]
word_count = word_choice.length()
jumble = ""
word_choice_arr = word_choice.split("")
jumbled_arr = []
while word_count != 0
	position = rand(word_count)
	jumbled_arr << word_choice_arr[position]
	word_choice_arr.delete_at(position)
	word_count -= 1
end
puts "The word is #{word_choice}"
puts "The jumbled word is #{jumbled_arr.join()}"

guess_count = 5
while guess_count != 0
	puts "Does the word contain the letter: "
	letter_guess = gets().chomp()
	if word_choice.include?(letter_guess)
		puts "Yes"
		guess_count -= 1
	else
		puts "No"
		guess_count -= 1
	end
end
puts "Guess the word: "
guess = gets().chomp()
while word_count == 0
	if guess == word_choice
		puts "Correct!"
		break
	else
		puts "Incorrect."
		puts "Guess the word: "
		guess = gets().chomp()
	end
end

